owrx/websocket.py

```diff
@@ -175,60 +177,67 @@ def read_loop(self):
         def protected_read(num):
             data = self.handler.rfile.read(num)
             if data is None:
-                raise Drained()
-            if len(data) != num:
+                data = bytes()
+            while self.open and len(data) < num:
+                    (read, _, _) = select.select([self.interruptPipeRecv, self.handler.rfile], [], [], 15)
+                    if self.handler.rfile in read:
+                        data += self.handler.rfile.read(num - len(data))
+                    else:
+                        if len(data) == 0:
```

Comment:
The else branch comes into effect if the interrupt pipe has been triggered, so typically there's no need to check the data length or have different types of exceptions here.